addon_service/abstract/authorized_account/serializers.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+from django.core.exceptions import ValidationError as ModelValidationError
+from rest_framework_json_api import serializers
+
+from addon_service.common.credentials_formats import CredentialsFormats
+from addon_service.osf_models.fields import encrypt_string
+from addon_service.serializer_fields import (
+    CredentialsField,
+    EnumNameMultipleChoiceField,
+)
+from addon_toolkit import AddonCapabilities
+
+
+if TYPE_CHECKING:
+    from addon_service.abstract.authorized_account.models import AuthorizedAccount
+    from addon_service.models import ExternalStorageService
+
+REQUIRED_FIELDS = frozenset(["url", "account_owner", "authorized_operations"])
+
+
+class AuthorizedAccountSerializer(serializers.HyperlinkedModelSerializer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Check whether subclasses declare all of required fields
+        if not REQUIRED_FIELDS.issubset(set(self.fields.keys())):
+            raise Exception(
+                f"{self.__class__.__name__} requires {self.REQUIRED_FIELDS} to be instantiated"
+            )
```

Comment:
honest question: These are already in the Meta's `fields`, won't that blow up instantiation for is if they're not present?

Reply:
No, it won't, tried this in interpreter. Fields in Meta are desingned to constrain inferring fields from model
![image](https://github.com/user-attachments/assets/f6509d28-90a9-458d-861b-8c88c0f89ddc)